mlir/lib/Dialect/Rock/Transforms/LayoutEmitter.cpp

```diff
@@ -0,0 +1,64 @@
+//===- LayoutEmitter.cpp - MLIR helper that contains the layout logic -===//
+//
+// Copyright 2020 The MLIR Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// =============================================================================
+//
+// This class tries to abstract away the code-generation details needed to
+// generated calls to matrix multiply accelerator intrinsics (wmma, mfma).
+//
+//
+//===----------------------------------------------------------------------===//
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/Rock/utility/AmdArchDb.h"
+#include "mlir/Dialect/Rock/utility/loweringUtils.h"
+
+#include "llvm/Support/Debug.h"
+
+#include "LayoutEmitter.h"
+
+#define DEBUG_TYPE "layout-transform"
```

Comment:
nit (naming) : "rock-grid-layout-emitter" using this in CLI with -debug-only="layout-transform" might be a little too vague.